### ISSUE TITLE
[BUG] [Formatter] --reformat is not idempotent: single-line {% if %}…{% endif %} blocks expand one-per-pass (regression in 1.39.0)

### ISSUE THREAD

**Issue by databyte** [djlint/djLint#2183]
- [x] I'm on the [latest version](https://pypi.org/project/djlint/) of djLint
- [x] I've searched the [issues](https://github.com/djlint/djLint/issues)
- [x] I've read the [docs](https://djlint.com)

## System Info

- OS: macOS 26.5.1 (arm64)
- Python Version: 3.14.3
- djLint Version: 1.40.2 (regression first appears in 1.39.0)
- template language: django

## Issue

`djlint --reformat` is **not idempotent** on templates that contain several single-line `{% if ... %}<tag>...</tag>{% endif %}` blocks. The formatter wants to expand each block onto multiple lines, but it only expands **one block per pass**. As a result, a single `--reformat` run leaves the file in a state that the next `--reformat` run changes again, and it takes as many passes as there are blocks before it stabilizes.

This breaks the standard "format once, then verify formatting" contract used by pre-commit hooks and CI: immediately after `djlint --reformat`, `djlint --check` still reports the file needs formatting.

This is a **regression introduced in 1.39.0** — 1.38.0 and earlier reach a stable result in a single pass. It appears related to the 1.39.0 change *"Preserved multiline control-flow blocks."*

## How To Reproduce

Save this as `contact_form.html` (a real-world Django form; the behaviour needs the full-form context — a single field in isolation does not reproduce it):

```html
<form id="contact-form"
      class="fade-rise glass rounded-3xl p-8"
      hx-post="{% url 'contact_submit' %}"
      hx-target="#contact-form"
      hx-swap="outerHTML"
      hx-disabled-elt="find button[type=submit]">
  {% csrf_token %}
  {% if form.non_field_errors %}
    <div class="mb-4 rounded-xl border border-red-300 bg-red-50 px-4 py-3 text-sm text-red-800"
         role="alert">{{ form.non_field_errors }}</div>
  {% endif %}
  <div class="grid gap-4 sm:grid-cols-2">
    <div>
      <input name="first_name"
             value="{{ form.first_name.value|default_if_none:'' }}"
             placeholder="First Name"
             autocomplete="given-name"
             class="w-full rounded-xl border border-gray-300 bg-white px-4 py-3 text-gray-900 shadow-sm placeholder:text-gray-400 focus:border-swift-500 focus:ring-swift-500">
      {% if form.first_name.errors %}<p class="mt-1 text-xs text-red-600">{{ form.first_name.errors.0 }}</p>{% endif %}
    </div>
    <div>
      <input name="last_name"
             value="{{ form.last_name.value|default_if_none:'' }}"
             placeholder="Last Name"
             autocomplete="family-name"
             class="w-full rounded-xl border border-gray-300 bg-white px-4 py-3 text-gray-900 shadow-sm placeholder:text-gray-400 focus:border-swift-500 focus:ring-swift-500">
      {% if form.last_name.errors %}<p class="mt-1 text-xs text-red-600">{{ form.last_name.errors.0 }}</p>{% endif %}
    </div>
    <div class="sm:col-span-2">
      <input name="email"
             type="email"
             value="{{ form.email.value|default_if_none:'' }}"
             placeholder="Business Email"
             autocomplete="email"
             class="w-full rounded-xl border border-gray-300 bg-white px-4 py-3 text-gray-900 shadow-sm placeholder:text-gray-400 focus:border-swift-500 focus:ring-swift-500">
      {% if form.email.errors %}<p class="mt-1 text-xs text-red-600">{{ form.email.errors.0 }}</p>{% endif %}
    </div>
    <div class="sm:col-span-2">
      <input name="job_title"
             value="{{ form.job_title.value|default_if_none:'' }}"
             placeholder="Job Title"
             autocomplete="organization-title"
             class="w-full rounded-xl border border-gray-300 bg-white px-4 py-3 text-gray-900 shado0 focus:border-swift-500 focus:ring-swift-500">
      {% if form.job_title.errors %}<p class="mt-1 text-xs text-red-600">{{ form.job_title.errors.0 }}</p>{% endif %}
    </div>
  </div>
</form>
```

Run (using the same options as our config: `--profile django --indent 2 --max-line-length 120`):

```bash
# 1) reformat once, then ask whether it is still formatted:
djlint contact_form.html --reformat --profile django --indent 2 --max-line-length 120
djlint contact_form.html --check   --profile django --indent 2 --max-line-length 120
```

**Actual output (1.39.0, 1.40.x):** the `--check` right after `--reformat` reports:

```
1 file would be updated.
```

Each successive `--reformat` expands one more `{% if %}...{% endif %}` block; the file only stabilises after ~4 passes.

**Expected output:** `0 files would be updated.` — a single `--reformat` should produce a fully-for

## Version bisect

| Version | Result |
|---|---|
| 1.38.0 and earlier | ✅ stable after one `--reformat` |
| **1.39.0** | ❌ non-idempotent (one block expanded per pass) |
| 1.40.0 / 1.40.1 / 1.40.2 | ❌ still non-idempotent |

```bash
uvx djlint@1.38.0 contact_form.html --reformat … && uvx djlint@1.38.0 contact_form.html --check …
uvx djlint@1.39.0 contact_form.html --reformat … && uvx djlint@1.39.0 contact_form.html --check …   # 1 file would be updated
```

**Comment by monosans** [comment]
Fixed in https://github.com/djlint/djLint/commit/655afef790db5a0a5f2cd3b3382247f8f129d885

**Comment by monosans** [comment]
Thanks for nicely detailed issues!
